backbone/src/main/java/org/researchstack/backbone/ResourceManager.java

```diff
@@ -113,4 +122,53 @@ public static ResourceManager getInstance() {
      * Onboarding sections
      */
     public abstract Resource getOnboardingManager();
+
+    /**
+     * The NameJsonProvider is a useful class that utilizes the
+     * ResourceManager to convert simple String names to Json Strings
+     */
+    public static class NameJsonProvider implements ResourceNameToStringConverter {
+
+        private Context context;
+
+        public NameJsonProvider(Context context) {
+            this.context = context;
+        }
+
+        @Override
+        public String getJsonStringForResourceName(String resourceName) {
```

Comment:
I'm glad you called this out and I was able to take a second look at it.  It does seem like I was trying to make the ResourceManager more open-ended so that all you needed was a filename and you could get a path to a file, instead of having to add an abstract method every time one is needed.  It seems like a better approach is to refactor ResourceManager so that it can be more extensible.  I'll work on that before merging in this PR.